statsmodels/tsa/vecm/vecm.py

```diff
@@ -0,0 +1,2146 @@
+
+        Combinations of these are possible (e.g. "cili" or "colo" for linear
+        trend with intercept)
+    seasons : int
+        Number of seasons.
+
+    Returns
+    -------
+    num : int
+        Number of deterministic terms and number dummy variables for seasonal
+        terms.
+    """
+    num = 0
+    if "ci" in det_string or "co" in det_string:
+        num += 1
+    if "li" in det_string or "lo" in det_string:
+        num += 1
+    if seasons > 0:
+        num += seasons - 1
+    return num
+
+
+def deterministic_to_exog(deterministic, seasons, nobs_tot, first_season=0,
+                          seasons_centered=False, exog=None, exog_coint=None):
+    """
+    Use the VECM's deterministic terms to construct an array that is suitable
+    to convey this information to VAR in form of the exog-argument for VAR's
+    __init__ method.
+
+    Parameters
+    ----------
+    deterministic : str
+        See VECM's docstring for more information.
+    seasons : int
+        Number of seasons.
+    nobs_tot : int
+        Number of observations including the presample.
+    first_season : int, default: 0
+        Season of the first observation.
+    seasons_centered : boolean, default: False
+        If True, the seasonal dummy variables are demeaned such that they are
+        orthogonal to an intercept term.
+    exog : ndarray (nobs_tot x #det_terms) or None, default: None
+        An ndarray representing deterministic terms outside the cointegration
+        relation.
+    exog_coint : ndarray (nobs_tot x #det_terms_coint) or None, default: None
+        An ndarray representing deterministic terms inside the cointegration
+        relation.
+
+    Returns
+    -------
+    exog : ndarray or None
+        None, if the function's arguments don't contain deterministic terms.
+        Otherwise, an ndarray representing these deterministic terms.
+    """
+    exogs = []
+    if "co" in deterministic or "ci" in deterministic:
+        exogs.append(np.ones(nobs_tot))
+    if exog_coint is not None:
+        exogs.append(exog_coint)
+    if "lo" in deterministic or "li" in deterministic:
+        exogs.append(np.arange(nobs_tot))
+    if seasons > 0:
+        exogs.append(seasonal_dummies(seasons, nobs_tot,
+                                      first_period=first_season,
+                                      centered=seasons_centered))
+    if exog is not None:
+        exogs.append(exog)
+    return np.column_stack(exogs) if exogs else None
+
+
+def _mat_sqrt(_2darray):
+    """Calculates the square root of a matrix.
+
+    Parameters
+    ----------
+    _2darray : ndarray
+        A 2-dimensional ndarray representing a square matrix.
+
+    Returns
+    -------
+    result : ndarray
+        Square root of the matrix given as function argument.
+    """
+    u_, s_, v_ = svd(_2darray, full_matrices=False)
+    s_ = np.sqrt(s_)
+    return chain_dot(u_, np.diag(s_), v_)
```

Comment:
s_ should broadcasts. I use this shortcut all the time with diagonal matrices. I just need to check each time whether it needs to be row or column s_ or s_[:, None]